components/eam/src/physics/crm/crm_physics.F90

```diff
@@ -673,50 +661,41 @@ subroutine crm_physics_tend(ztodt, state, tend, ptend, pbuf, cam_in, cam_out, &
    !------------------------------------------------------------------------------------------------
    ! Set CRM orientation angle
    !------------------------------------------------------------------------------------------------
-   crm_angle(:) = 0
+   if (MMF_orientation_angle==-1) then
 
-#if defined( MMF_ORIENT_RAND )
-   !------------------------------------------------------------------------------------------------
-   ! Rotate the CRM using a random walk
-   !------------------------------------------------------------------------------------------------
-   if ( (crm_ny.eq.1) .or. (crm_nx.eq.1) ) then
+      crm_rotation_std    = 20. * pi/180.                 ! std deviation of normal distribution for CRM rotation [radians]
+      crm_rotation_offset = 90. * pi/180. * ztodt/86400.  ! This means that a CRM should rotate 90 deg / day on average
 
-      if (.not. is_first_step()) then
-         ! get current crm angle from pbuf, except on first step
-         call pbuf_get_field(pbuf, pbuf_get_index('CRM_ANGLE'), crm_angle)
+      ! Rotate the CRM using a random walk
+      if ( (crm_ny.eq.1) .or. (crm_nx.eq.1) ) then
+         if (.not. is_first_step()) then
+            ! get current crm angle from pbuf, except on first step
+            call pbuf_get_field(pbuf, crm_angle_idx, crm_angle_ptr)
+            crm_angle(:pcols) = crm_angle_ptr(:pcols)
```

Comment:
I don't think it is wrong but I am just curious why you decided to use a pointer, crm_angle_ptr, instead of just using crm_angle inside pbuf_get_field.

Reply:
Yea, I guess I can do it like that. I had done it this way for another branch because passing pointers to the CRM is problematic given how it's been written for the GPU, but in this case the variable is only used locally. I'll switch to just using a pointer to make the code a bit simpler.